specification.md

```diff
@@ -2074,9 +2093,35 @@ This field can be used to perform a client certificate revocation/rotation.
 Other connection settings. These are Agent-specific and are up to the Agent
 interpret.
 
-#### Headers Message
+##### OtherConnectionSettings.tls
+
+Status: [Development]
 
+Optional connection specific TLS settings.
+
+#### TLSConnectionSettings Message
+
+Status: [Development]
+
+The message carries optional TLS settings that are used to configure a client's
+connection. If the Agent is able to validate the connection settings, the Agent
+SHOULD forget any previous TLS settings. If this message is not included, the
+client SHOULD use the agent's default TLS settings for the connection.
```

Comment:
@tigrannajaryan @andykellr, I've added a description on how TLS settings should function, and have changed the demo (https://github.com/open-telemetry/opamp-go/pull/338) to reflect this.

The goal is that when `TLSConnectionSettings` is present, it can form the TLS config (excluding client certificate).
If no TLS settings are sent with a connection offering, the agent's defaults should be used.